bookkeeper-server/src/main/java/org/apache/bookkeeper/client/LedgerHandle.java

```diff
@@ -760,6 +798,106 @@ public CompletableFuture<LedgerEntries> readAsync(long firstEntry, long lastEntr
         return readEntriesInternalAsync(firstEntry, lastEntry, false);
     }
 
+    @Override
+    public CompletableFuture<LedgerEntries> readAsync(long startEntry, int maxCount, long maxSize) {
```

Comment:
We shouldn't provide the fallback logic in the normal read method. We should provide a new method named with fallback. https://github.com/apache/bookkeeper/pull/4051/files#diff-4a1b643197f23e320d89550d6ae641179955e03f61332c69a82e10cacaa12390R62